samples/dnn/speech_recognition.py

```diff
@@ -0,0 +1,454 @@
+                hop_length : number of audio samples between adjacent STFT columns.
+                win_length : Each frame of audio is windowed by window of length win_length and
+                    then padded with zeros to match n_fft
+                fft_window : a vector or array of length `n_fft` having values computed by a
+                    window function
+                pad_mode : mode while padding the singnal
+                return_complex : returns array with complex data type if `True`
+            return : Matrix of short-term Fourier transform coefficients.
+        '''
+        if win_length is None:
+            win_length = n_fft
+        if hop_length is None:
+            hop_length = int(win_length // 4)
+        if y.ndim!=1:
+            raise Exception(f'Invalid input shape. Only Mono Channeled audio supported. Input must have shape (Audio,). Got {y.shape}')
+
+        # Pad the window out to n_fft size
+        fft_window = self.pad_window_center(fft_window, n_fft)
+
+        # Reshape so that the window can be broadcast
+        fft_window = fft_window.reshape((-1, 1))
+
+        # Pad the time series so that frames are centered
+        y = np.pad(y, int(n_fft // 2), mode=pad_mode)
+
+        # Window the time series.
+        y_frames = self.frame(y, frame_length=n_fft, hop_length=hop_length)
+
+        # Convert data type to complex
+        dtype = self.dtype_r2c(y.dtype)
+
+        # Pre-allocate the STFT matrix
+        stft_matrix = np.empty( (int(1 + n_fft // 2), y_frames.shape[-1]), dtype=dtype, order="F")
+
+        stft_matrix = np.fft.rfft( fft_window * y_frames, axis=0)
+        return stft_matrix if return_complex==True else np.stack((stft_matrix.real,stft_matrix.imag),axis=-1)
+
+class Decoder():
+    '''
+        Used for decoding the output of jasper model.
+    '''
+    def __init__(self):
+        labels=[' ','a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z',"'"]
+        self.labels_map = {i: labels[i] for i in range(28)}
+        self.blank_id = 28
+
+    def decode(self,x):
+        """
+            Takes output of Jasper model and performs ctc decoding algorithm to
+            remove duplicates and special symbol. Returns prediction
+        """
+        x = np.argmax(x,axis=-1)
+        hypotheses = []
+        prediction = x.tolist()
+        # CTC decoding procedure
+        decoded_prediction = []
+        previous = self.blank_id
+        for p in prediction:
+            if (p != previous or previous == self.blank_id) and p != self.blank_id:
+                decoded_prediction.append(p)
+            previous = p
+        hypothesis = ''.join([self.labels_map[c] for c in decoded_prediction])
+        hypotheses.append(hypothesis)
+        return hypotheses
+
+if __name__ == '__main__':
+
+    # Computation backends supported by layers
+    backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_OPENCV)
```

Comment:
Could you try forward net with OpenVINO (cv.dnn.DNN_BACKEND_INFERENCE_ENGINE)?

Reply:
I tried. It gave this error: `error: (-213:The function/feature is not implemented) Unknown backend identifier in function 'cv::dnn::dnn4_v20210301::wrapMat'`